### ISSUE TITLE
Connecting to an local network instance of ollama fails

### ISSUE THREAD

**Issue by DIGist** [rmusser01/tldw#388]
**Are You on the Latest version?**
You did a git pull and are running the latest version/build?
Y

**Please describe the bug**
attempting to connect to a network instance of ollama, it still seems to be trying to control the llama instance through ollama cli
config.txt:
ollama_api_IP = http://myserver:33821/v1/chat/completions
ollama_api_key = 411
ollama_model = qwen2.5

running:
python summarize.py -gui -api ollama --local_llm
tried with and without the --local_llm switch

log:

> INFO:root:Local LLM flag: True
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 12 arguments for function <function import_file_handler at 0x7fb2dd96c540>, received 11.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 12 arguments for function <function import_file_handler at 0x7fb2dd96c540>, received 11.
>   warnings.warn(
> INFO:App_Function_Libraries.DB.RAG_QA_Chat_DB:Retrieved page 1 of 0 (total items: 0)
> INFO:App_Function_Libraries.DB.RAG_QA_Chat_DB:Retrieved 0 conversations (page 1 of 0)
> INFO:App_Function_Libraries.DB.RAG_QA_Chat_DB:Retrieved page 1 of 0 (total items: 0)
> INFO:App_Function_Libraries.DB.RAG_QA_Chat_DB:Retrieved 0 conversations (page 1 of 0)
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 14 arguments for function <function create_rag_qa_chat_tab.<locals>.rag_qa_chat_wrapper at 0x7fb2d865dd00>, received 13.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 14 arguments for function <function create_rag_qa_chat_tab.<locals>.rag_qa_chat_wrapper at 0x7fb2d865dd00>, received 13.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 3 arguments for function <function save_chat_history at 0x7fb2da6696c0>, received 2.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 3 arguments for function <function save_chat_history at 0x7fb2da6696c0>, received 2.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 1 arguments for function <function show_edit_message at 0x7fb2da6684a0>, received 0.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 1 arguments for function <function show_edit_message at 0x7fb2da6684a0>, received 0.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 1 arguments for function <function show_delete_message at 0x7fb2da668d60>, received 0.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 1 arguments for function <function show_delete_message at 0x7fb2da668d60>, received 0.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 4 arguments for function <function create_character_card_interaction_tab.<locals>.import_chat_history at 0x7fb2d84ed6c0>, received 3.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 4 arguments for function <function create_character_card_interaction_tab.<locals>.import_chat_history at 0x7fb2d84ed6c0>, received 3.
>   warnings.warn(
> {"asctime": "2024-10-22 06:59:30,257", "levelname": "INFO", "name": "tldw_app_logs", "event": "load_characters_attempt", "type": "counter", "value": 1, "labels": {}, "timestamp": "2024-10-22T06:59:30.257466+00:00Z", "taskName": null}
> INFO:tldw_app_logs:metric
> WARNING:root:Characters file not found: /tldw/App_Function_Libraries/../Helper_Scripts/Character_Cards/Characters.json
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1002: UserWarning: Expected 5 arguments for function <function adjust_tone at 0x7fb2da694fe0>, received 3.
>   warnings.warn(
> /tldw/venv/lib/python3.12/site-packages/gradio/utils.py:1006: UserWarning: Expected at least 5 arguments for function <function adjust_tone at 0x7fb2da694fe0>, received 3.
>   warnings.warn(
> Traceback (most recent call last):
>   File "/tldw/summarize.py", line 861, in <module>
>     launch_ui(share_public=False)
>   File "/tldw/App_Function_Libraries/Gradio_Related.py", line 388, in launch_ui
>     create_ollama_tab()
>   File "/tldw/App_Function_Libraries/Local_LLM/Local_LLM_ollama.py", line 66, in create_ollama_tab
>     model_list = gr.Dropdown(label="Available Models", choices=get_ollama_models())
>                                                                                                                  ^^^^^^^^^^^^^^^^^^^
>   File "/tldw/App_Function_Libraries/Local_LLM/Local_LLM_ollama.py", line 12, in get_ollama_models
>     result = subprocess.run(['ollama', 'list'], capture_output=True, text=True, check=True)
>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/subprocess.py", line 548, in run
>     with Popen(*popenargs, **kwargs) as process:
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/subprocess.py", line 1026, in __init__
>     self._execute_child(args, executable, preexec_fn, close_fds,
>   File "/usr/lib/python3.12/subprocess.py", line 1955, in _execute_child
>     raise child_exception_type(errno_num, err_msg, err_filename)
> FileNotFoundError: [Errno 2] No such file or directory: 'ollama'
> 

**Is the bug reproducable reliably?**
Yes

**What was the expected behavior?**
connect to a networked ollama instance

**Comment by DIGist** [comment]
@rmusser01 Thanks! I got around to trying it again, it's working with my networked ollama instance when just running with "python summarize.py -gui" and configuring the ollama values in the config of the web ui.

Thanks! Now I just need to figure out how to get diarization to work, and I'll be rolling!

**Comment by rmusser01** [comment]
Thank you for the follow up!
Yea, diarization is pretty wonky right now, admittedly I struggled with figuring out how to do it without having to call out to HF, figured it out, tested it, and found it to be pretty bad. 
Figured it was probably my configuration of the pipeline, but didn't really care to dive into it too much at the time, as I felt other features were higher priority.

I'm planning to revisit the existing transcription pipeline in probably a week? (faster_whisper implemented some new features which should improve processing times by _a lot_) at which point I'll dive back into the diarization and figure out why its crappy.

**Comment by DIGist** [comment]
@rmusser01 Thanks! Good to know, this collection of features has been something i've been dreaming of, but don't have the time to put together, so I appreciate your work. I only got as far as using whisperx with diarization to create transcripts. The diarization was nutty too there, cause when I did it you had to configure a hugging face key and signin to huggingface to agree to the terms for the diarization model for it it be able to pull it and work.
